### PR TITLE
Refactor handlers & database/datastore, add handler tests

### PR DESCRIPTION
**Pull request by smartin015**
This attempts to make the code more functional and testable.

- Database is passed instead of individual models (keeps the interface light)
- Added testing utilities for quickly setting up & populating an inmemory DB
- Split "mail" code out into helper methods to make the logic more clear
- Moved all Sequelize model setup stuff into Database.ts
- Turned methods from original model classes into top-level helper methods
- Plumbed these changes through the codebase.
- Tested all db-using handlers, and most of the others

### CODE REVIEW

File: app/Handlers.ts [hunk 1]
```diff
@@ -171,39 +174,43 @@ export function publish(quest: Quest, req: express.Request, res: express.Respons
     contentrating: req.query.contentrating,
     expansionhorror: req.query.expansionhorror || false,
     language: req.query.language,
-  };
+  });
+  if (quest instanceof Error) {
+    console.error('HERP');
```
Comment (toddmedema) [inline app/Handlers.ts]: Lol, maybe `'HERP', quest`? :)

File: app/Routes.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import * as express from 'express'
 import Config from './config'
 import * as Mail from './Mail'
-import * as oauth2 from './lib/oauth2'
+import {template as oauth2Template, installOAuthRoutes} from './lib/oauth2'
```
Comment (toddmedema) [inline app/Routes.ts]: Could we have the oauth lib just export oauth2Template directly?

File: app/models/TestData.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    questID: 'questid',
+    userID: 'userid',
+    questVersion: 1,
+    difficulty: 'NORMAL',
+    platform: 'ios',
+    players: 5,
+    version: '1.0.0',
+  }),
+  action: new AnalyticsEvent({
+  category: 'category',
```
Comment (toddmedema) [inline app/models/TestData.ts]: Missing inentation - shouldn't the linter have caught this?

File: app/models/TestData.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  name: 'Test Testerson',
+  difficulty: 'NORMAL',
+  platform: 'ios',
+  players: 5,
+  version: '1.0.0',
+  anonymous: false,
+  tombstone: PLACEHOLDER_DATE,
+});
+export const feedback = {
+  basic: basicFeedback,
```
Comment (toddmedema) [inline app/models/TestData.ts]: Why'd we switch to abstracting for just this one?

... ahh, to reduce redundancy & maintenance with spreads. Perhaps worth also doing to the other test examples?

File: app/Handlers.ts (hunk 1 above)
Comment (smartin015) [reply app/Handlers.ts]: Done.

File: app/Routes.ts (hunk 2 above)
Comment (smartin015) [reply app/Routes.ts]: Done.

File: app/models/TestData.ts (hunk 3 above)
Comment (smartin015) [reply app/models/TestData.ts]: Looks like the linter only catches odd spacing, or tabs. We could use something like [prettier](https://github.com/prettier/prettier) for stricter formatting if we wanted.

Fixed this problem at least.

File: app/models/TestData.ts (hunk 4 above)
Comment (smartin015) [reply app/models/TestData.ts]: Done.

**Pull request merged**